app/main.py

```diff
@@ -1,3 +1,37 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+
+from app.customer import Customer
+from app.car import Car
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    customers = []
+    shops = []
+    with open("app/config.json") as file:
+        data = json.load(file)
+        Car.fuel_price = data["FUEL_PRICE"]
+        for customer in data["customers"]:
+            customers.append(
+                Customer(
+                    customer["name"],
+                    customer["product_cart"],
+                    customer["location"],
+                    customer["money"],
+                    Car(
+                        customer["car"]["brand"],
+                        customer["car"]["fuel_consumption"]
+                    )
+                )
+            )
+        for shop in data["shops"]:
+            shops.append(
+                Shop(
+                    shop["name"],
+                    shop["location"],
+                    shop["products"]
+                )
+            )
+
```

Comment:
Do not overload the context manager here, use it only to get data from the file, and make the other logic outside it.